src/Microsoft.NetCore.Analyzers/Core/Security/SourceTriggeredTaintedDataAnalyzerBase.cs

```diff
@@ -77,13 +83,81 @@ public override void Initialize(AnalysisContext context)
                                 operationAnalysisContext =>
                                 {
                                     IInvocationOperation invocationOperation = (IInvocationOperation)operationAnalysisContext.Operation;
-                                    if (sourceInfoSymbolMap.IsSourceMethod(invocationOperation.TargetMethod))
+                                    bool needValueContentAnalysis = false;
+                                    if (sourceInfoSymbolMap.IsSourceMethod(invocationOperation.TargetMethod, out var evaluateWithPointsToAnalysis, out var evaluateWithValueContentAnslysis))
                                     {
-                                        rootOperationsNeedingAnalysis.Add(operationAnalysisContext.Operation.GetRoot());
+                                        IOperation rootOperation = operationAnalysisContext.Operation.GetRoot();
+                                        PointsToAnalysisResult pointsToAnalysisResult;
+                                        ValueContentAnalysisResult valueContentAnalysisResultOpt;
+                                        if (evaluateWithPointsToAnalysis.Count != 0)
+                                        {
+                                            pointsToAnalysisResult = PointsToAnalysis.TryGetOrComputeResult(
+                                                rootOperation.GetEnclosingControlFlowGraph(),
+                                                owningSymbol,
+                                                WellKnownTypeProvider.GetOrCreate(operationAnalysisContext.Compilation),
+                                                InterproceduralAnalysisConfiguration.Create(
+                                                    operationAnalysisContext.Options,
+                                                    SupportedDiagnostics,
+                                                    defaultInterproceduralAnalysisKind: InterproceduralAnalysisKind.ContextSensitive,
+                                                    cancellationToken: operationAnalysisContext.CancellationToken),
+                                                interproceduralAnalysisPredicateOpt: null);
+                                            if (pointsToAnalysisResult == null
```

Comment:
I probably told you that DFA results are usually cached, but it turns out I was wrong.  If you rerun this line in the debugger, you get back different instances of PointsToAnalysisResults.

So for now, I'd say just `IsSourceMethod()` returning back true is good enough to know we need to perform tainted data analysis.

We can revisit after we optimize DFA results caching.

Reply:
Actually, I think I'll just fix the DFA caching so it does work.  🙂  See #2660

Reply:
Merged, so nevermind!